src/main/scala/za/co/absa/commons/reflect/FieldValueExtractor.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.commons.reflect
+
+import za.co.absa.commons.annotation.DeveloperApi
+import za.co.absa.commons.reflect.ReflectionUtils.allInterfacesOf
+
+import scala.annotation.tailrec
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe.{Mirror, runtimeMirror}
+
+@DeveloperApi
+class FieldValueExtractor[A: ClassTag, B](o: AnyRef, fieldName: String) {
```

Comment:
Does it need to a class? It doesn't seem to have any state, so shouldn't it rather be an `object` ?

Reply:
I was trying to reduce the nesting, by defining the o and fieldName as constructor parameters all the methods can be on top level.